packages/storybook/src/index.ts

```diff
@@ -86,5 +93,94 @@ export async function argosScreenshot(
         : {}),
     },
     ...fitToContentOptions,
+  };
+
+  const [version, storyContext] = await Promise.all([
+    getArgosStorybookVersion(),
+    getStoryContext(page, context),
+  ]);
+
+  DO_NOT_USE_setMetadataConfig({
+    sdk: { name: "@argos-ci/storybook", version },
+    playwrightLibraries: ["@storybook/test-runner"],
   });
+
+  const argosParameters = getArgosParameters(storyContext.parameters);
+  const modes = argosParameters?.modes;
+
+  if (modes) {
+    for (const [name, globals] of Object.entries(modes)) {
+      // Skip disabled modes.
+      if (globals.disabled) {
+        continue;
+      }
+      await setStorybookGlobals({ page, globals });
+      await stabilizeAndScreenshot({
+        page,
+        context,
+        options: argosOptions,
+        suffix: ` ${name}`,
+      });
+    }
+  } else {
+    await stabilizeAndScreenshot({ page, context, options: argosOptions });
+  }
+}
+
+/**
+ * Set the Storybook globals.
+ */
+async function setStorybookGlobals(args: {
+  page: Page;
+  globals: StorybookGlobals;
+}) {
+  const { page, globals } = args;
+  await page.evaluate((globals) => {
+    (globalThis as any).__STORYBOOK_PREVIEW__.channel.emit("updateGlobals", {
+      globals,
+    });
+  }, globals);
+}
+
+/**
+ * Wait for the page to be ready and take a screenshot.
+ */
+async function stabilizeAndScreenshot(args: {
+  page: Page;
+  context: TestContext;
+  options: ArgosScreenshotOptions;
+  suffix?: string;
+}) {
+  const { page, context, options } = args;
+  await waitForPageReady(page);
+  return argosPlaywrightScreenshot(
+    page,
+    join(context.title, context.name) + (args.suffix ?? ""),
+    options,
+  );
+}
+
+/**
+ * Get the Argos parameters from the Storybook context.
+ */
+function getArgosParameters(
```

Comment:
```suggestion
function getParameters(
```

Reply:
It takes parameters in input and returns `parameters.argos`, so it's `getArgosParameters`.